rust/agama-dbus-server/src/network/nm/dbus.rs

```diff
@@ -159,6 +173,48 @@ fn wireless_config_to_dbus(conn: &WirelessConnection) -> NestedHash {
     ])
 }
 
+/// Converts a MatchConfig struct into a HashMap that can be sent over D-Bus.
+///
+/// * `match_config`: MatchConfig to convert.
+fn match_config_to_dbus(match_config: &MatchConfig) -> HashMap<&str, zvariant::Value> {
+    let mut match_config_dbus = HashMap::new();
+
+    let drivers: Value = match_config
+        .driver
+        .iter()
+        .map(|dr| dr.to_string())
+        .collect::<Vec<String>>()
+        .into();
+    match_config_dbus.insert("driver", drivers);
+
+    let kernels: Value = match_config
+        .kernel
+        .iter()
+        .map(|dr| dr.to_string())
+        .collect::<Vec<String>>()
+        .into();
+    match_config_dbus.insert("kernel-command-line", kernels);
+
+    let paths: Value = match_config
+        .path
+        .iter()
+        .map(|dr| dr.to_string())
+        .collect::<Vec<String>>()
+        .into();
+    match_config_dbus.insert("path", paths);
+
+    let interfaces: Value = match_config
+        .interface
+        .iter()
+        .map(|dr| dr.to_string())
+        .collect::<Vec<String>>()
+        .into();
+
+    match_config_dbus.insert("interface-name", interfaces);
+
+    match_config_dbus
```

Comment:
You can do:

```rust
    HashMap::from([
        ("driver", drivers),
        ("kernel-command-line", kernels),
        ("path", paths),
        ("interface-name", interfaces),
    ])
```

It looks more readable to me and it is easy to find out what is included in the HashMap.

Reply:
Yep, originally did so, then added some logic to inserting only the key if it was not empty but as we are inserting it always now it makes sense to do so. Thanks